### PR TITLE
When using set_default_permissions OPTIONS calls for CORS does not work anymore.

### PR DESCRIPTION
**Pull request by Natim**
Because browser doesn't send Authorization headers on OPTIONS call.

### CODE REVIEW

File: cornice/pyramidhook.py [hunk 1]
```diff
@@ -159,7 +159,8 @@ def register_service_views(config, service):
     # before doing anything else, register a view for the OPTIONS method
     # if we need to
     if service.cors_enabled and 'OPTIONS' not in service.defined_methods:
-        service.add_view('options', view=get_cors_preflight_view(service))
+        service.add_view('options', view=get_cors_preflight_view(service),
+                         permission=NO_PERMISSION_REQUIRED)
```
Comment (ametaireau) [inline cornice/pyramidhook.py]: Good catch thanks!

**Comment by ametaireau** [comment]
Can you add a test to reflect that?

**Comment by Natim** [comment]
I tried to reproduce using a test but I wasn't able to with service. I will try with Resource.

**Comment by Natim** [comment]
Here we go :)

File: cornice/tests/test_cors.py [hunk 2]
```diff
@@ -98,12 +101,11 @@ def setUp(self):
         self.config = testing.setUp()
         self.config.include("cornice")
         self.config.add_route('noservice', '/noservice')
-
         self.config.scan("cornice.tests.test_cors")
         self.app = TestApp(CatchErrors(self.config.make_wsgi_app()))
 
-        def tearDown(self):
-            testing.tearDown()
+    def tearDown(self):
```
Comment (Natim) [inline cornice/tests/test_cors.py]: oops

File: cornice/tests/test_cors.py [hunk 3]
```diff
@@ -312,3 +314,36 @@ def test_existing_non_service_route(self):
         resp = self.app.get('/noservice', status=200,
                             headers={'Origin': 'notmyidea.org'})
         self.assertEqual(resp.body, b'No Service here.')
+
+
+class TestAuthenticatedCors(TestCase):
```
Comment (Natim) [inline cornice/tests/test_cors.py]: CORS, upper case

**Comment by Natim** [comment]
(@ametaireau speaking) that works for me. Changes to do: single to double quotes everywhere, update Cors in CORS and rename "noservice" in something else.

**Pull request merged**